### ISSUE TITLE
Suboptimal handling of enum arguments

### ISSUE THREAD

**Issue by pavel-kirienko** [madman-bob/python-argparse-utils#2]
First of all, I want to say that this is a very useful library and I am surprised that I am the only watcher of this repository. Lack of native support for enumerations in the built-in argparse module is quite unfortunate and this library provides a decent solution to that.

This is my argument configuration:

```python
    parser.add_argument(
        '--priority',
        default=pyuavcan.presentation.DEFAULT_PRIORITY,
        action=argparse_utils.enum_action(pyuavcan.transport.Priority),
        help='''
Priority of published message transfers. Applies to the heartbeat as well.
Default: %(default)s
'''.strip(),
    )
```

...where [`pyuavcan.transport.Priority` is an enumeration](https://github.com/UAVCAN/pyuavcan/blob/d348a81fbfb96350305e18bf2188deecadde01d6/pyuavcan/transport/_transfer.py#L19-L31). If the user specifies an incorrect input, the resulting error message is non-human-friendly at best:

> pyuavcan publish: error: argument --priority: invalid choice: 'EXCEPTIONA' (choose from <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafada0>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafadd8>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafae10>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafae48>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafae80>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafaeb8>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafaef0>, <argparse_utils.store_mapped_action.Choice object at 0x7f3f8eafaf28>)

Also, I would like enum names to be case-insensitive because uppercase command-line options are unpleasant to type and it feels weird. Further, it would be great if there was support for abbreviations, so that, for example, in the above case the library could match the prefix and recognize that the user tried to say `EXCEPTIONAL`.

Thanks!

**Comment by madman-bob** [comment]
Weird error message fixed in 1.3.0.

Also added in 1.3.0: `enum_action` now has an optional `key_normalizer` argument. Passing in `str.upper` for this argument will make your names case-insensitive. A custom, more complicated, `key_normalizer` would allow you to recognize abbreviations, also.

**Comment by pavel-kirienko** [comment]
Thank you, madman Bob :3
